test/catalog/catalog.test.ts

```diff
@@ -0,0 +1,202 @@
+    zeroMilk,
+} from '../shared';
+
+const genericItems: GenericTypedEntity[] = [genericCone];
+
+const specificItems: SpecificTypedEntity[] = [smallVanillaCone];
+
+const emptyGenericItems: GenericTypedEntity[] = [];
+
+const emptySpecificItems: SpecificTypedEntity[] = [];
+
+const testCatalog = Catalog.fromEntities(
+    genericItems.values(),
+    specificItems.values()
+);
+
+const emptyCatalog = Catalog.fromEntities(
+    emptyGenericItems.values(),
+    emptySpecificItems.values()
+);
+
+describe('Catalog', () => {
+    describe('merge', () => {
+        it('should be able to merge a catalog into an empty catalog', () => {
+            const catalog = Catalog.fromCatalog(emptyCatalog);
+            catalog.merge(testCatalog);
+            assert.deepEqual(catalog, testCatalog);
+        });
+
+        it('should be able to merge catalog into a full catalog', () => {
+            const testGeneric: GenericTypedEntity[] = [genericCoffee];
+            const testSpecific: SpecificTypedEntity[] = [
+                smallChocolateCone,
+                mediumCoffee,
+                zeroMilk,
+            ];
+            const catalog = Catalog.fromEntities(
+                testGeneric.values(),
+                testSpecific.values()
+            );
+
+            // Expected catalog items when we merge catalog with testCatalog
+            const expectedGeneric: GenericTypedEntity[] = [
+                genericCoffee,
+                genericCone,
+            ];
+            const expectedSpecific: SpecificTypedEntity[] = [
+                smallChocolateCone,
+                mediumCoffee,
+                zeroMilk,
+                smallVanillaCone,
+            ];
+            const expectedCatalog = Catalog.fromEntities(
+                expectedGeneric.values(),
+                expectedSpecific.values()
+            );
+
+            catalog.merge(testCatalog);
+            assert.deepEqual(catalog, expectedCatalog);
+        });
+
+        it('should throw an exception if there is a duplicate item key', () => {
+            const testGeneric: GenericTypedEntity[] = [genericCoffee];
+            const testSpecific: SpecificTypedEntity[] = [
+                smallChocolateCone,
+                smallVanillaCone,
+                mediumCoffee,
+                zeroMilk,
+            ];
+            const catalog = Catalog.fromEntities(
+                testGeneric.values(),
+                testSpecific.values()
+            );
+
+            expect(catalog.merge.bind(testCatalog)).to.throw(TypeError);
+        });
+    });
+
+    describe('hasPID', () => {
+        it('should return true if PID is found ', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            assert.isTrue(catalog.hasPID(genericConePID));
+        });
+        it('should return false if PID is not found', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            assert.isFalse(catalog.hasPID(genericCoffeePID));
```

Comment:
Is there already a PID with a name like unknownPID or nonexistantPID? This would make the test more readable than using genericCoffeePID.

Reply:
Did not realise UnknownPID and UnknownKey existed already - I'll switch to using them